libs/ui-components/src/components/Device/EditDeviceWizard/steps/ConfigInlineTemplateForm.tsx

```diff
@@ -1,20 +1,152 @@
+import { MinusCircleIcon } from '@patternfly/react-icons/dist/js/icons/minus-circle-icon';
+import { FieldArray, useField, useFormikContext } from 'formik';
 
 import { useTranslation } from '../../../../hooks/useTranslation';
-import { IgnitionFileHelperText } from '../../../common/HelperTextItems';
 import TextAreaField from '../../../form/TextAreaField';
+import FormSelectTypeahead from '../../../form/FormSelectTypeahead';
+import TextField from '../../../form/TextField';
+import { DeviceSpecConfigFormValues } from '../types';
+import { InlineConfigTemplate } from '../../../../types/deviceSpec';
+import ExpandableFormSection from '../../../form/ExpandableFormSection';
+import SwitchField from '../../../form/SwitchField';
+
+const FileForm = ({ fieldName }: { fieldName: string }) => {
+  const { t } = useTranslation();
+  const [{ value: file }] = useField<InlineConfigTemplate['files'][0]>(fieldName);
+
+  const permissions = React.useMemo(
+    () => ({
+      '777': t('(0777) Read, write, and execute permissions for all users.'),
+      '755': t("(0755) Read and execute permission for all users. The file's owner also has write permission."),
+      '750': t(
+        "(0750) Read and execute permission for the owner and group. The file's owner also has write permission. Users who aren't the file's owner or members of the group have no access to the file.",
+      ),
+      '700': t("(0700) Read, write, and execute permissions for the file's owner only; all others have no access."),
+      '666': t('(0666) Read and write permissions for all users. No execute permissions for anybody.'),
+      '664': t('(0664) Read and write permissions for the owner and group. Read- only permission for all others.'),
+      '660': t('(0660) Read and write permissions for the owner and group. No world permissions.'),
+      '644': t('(0644) Read and write permissions for the owner. Read-only permission for all others.'),
+      '640': t(
+        '(0640) Read and write permissions for the owner, and read-only permission for the group. No permission for others.',
+      ),
+      '600': t('(0600) Read and write permissions for the owner. No permission for anybody else.'),
+      '400': t('(0400) Read permission for the owner. No permission for anybody else.'),
+    }),
+    [t],
+  );
+
+  return (
+    <ExpandableFormSection title={t('File')} fieldName={fieldName} description={file.path}>
+      <Grid hasGutter>
+        <FormGroup label={t('File path on the device')} isRequired>
+          <TextField name={`${fieldName}.path`} />
+        </FormGroup>
+        <FormGroup label={t('Content')} isRequired>
+          <TextAreaField name={`${fieldName}.content`} />
+        </FormGroup>
+        <SwitchField label={t('Content is base64 encoded')} name={`${fieldName}.base64`} />
+        <FormGroup label={t('Permissions')}>
+          <FormSelectTypeahead
+            name={`${fieldName}.permissions`}
+            placeholderText={t('Select permissions')}
```

Comment:
According to the design, there should be a permanent helper text as well, that explains that permissions can be either selected or typed.